examples/hosts/app-integration/external-data/tests/mockCustomerService.test.ts

```diff
@@ -0,0 +1,71 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import type { Server } from 'http';
+
+import request from "supertest";
+
+import { initializeCustomerService } from '../src/mock-customer-service';
+import { customerServicePort } from '../src/mock-customer-service-interface';
+import { initializeExternalDataService } from '../src/mock-external-data-service';
+import { externalDataServicePort } from '../src/mock-external-data-service-interface';
+import { closeServer } from './utilities';
+
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+
+/**
+ * @remarks
+ *
+ * These tests spin up their own Express server instances so we can directly test against it
+ * (using supertest), rather than leaning on network calls.
+ */
+describe("mock-customer-service", () => {
+    /**
+     * Express server instance backing our mock external data service.
+     */
+    let externalDataService: Server | undefined;
+
+    /**
+     * Express server instance backing our mock customer service.
+     */
+    let customerService: Server | undefined;
+
+    beforeEach(async () => {
+        externalDataService = await initializeExternalDataService({
+            port: externalDataServicePort
+        });
+        customerService = await initializeCustomerService({
+            port: customerServicePort,
+            externalDataServiceWebhookRegistrationUrl: `http://localhost:${externalDataServicePort}/register-for-webhook`
+        });
+    });
+
+    afterEach(async () => {
+        const _externalDataService = externalDataService!;
+        const _customerService = customerService!;
+
+        externalDataService = undefined;
+        customerService = undefined;
+
+        await closeServer(_externalDataService);
+        await closeServer(_customerService);
```

Comment:
Do we need the local variables here? Why can't we directly do `closeServer(externalDataService!)` before un-assigning the variables?

Reply:
Technically that exposes a race condition, since we would be capturing the test properties in the promise callbacks, and other code could change the values out from under them. Our linter nicely complains about this.

Reply:
Oh, right, that's a tricky one. Thank you, linter; nice catch 😄! @jikim-msft @chentong7 FYI.